src/split.rs

```diff
@@ -38,37 +38,53 @@ use std::ffi::c_void;
 ///
 /// # Safety
 ///
-/// The returned function can only be called with the returned pointer, or a
-/// pointer to another `C` closure.
-pub unsafe fn split_closure<C, Args, Ret>(
-    closure: &mut C,
+///   - The returned function can only be called with the returned pointer, or a
+///     pointer to another `C` closure.
+///
+///   - Such call is only valid within the `'lifetime` of the borrow over the
+///     `closure`, during which such pointer cannot be aliased (_e.g._, no
+///     concurrent calls to the closure).
+///
+///   - The call must be performed within the same thread, unless `C` (_i.e._,
+///     the environment captured by the closure) is `Send`.
+pub fn split_closure<'lifetime, C, Args>(
```

Comment:
Looking back, I don't think we can actually do anything to make sure the *mut c_void` returned by this function doesn't outlive the `&'lifetime mut C` passed in. Is `'lifetime` just meant as a hint to the caller that they need to worry about lifetimes?

Reply:
Yes, kind of. It's so that the documention that mentions `'lifetime` leads to no ambiguity. I guess a wrapper pointer-with-lifetime type here would make sense indeed.